src/client/js/pie-img.js

```diff
@@ -40,3 +42,4 @@ export default class PieImg extends HTMLElement {
     }
   }
 }
+window.customElements.define('pie-img', PieImg);
```

Comment:
Yes, indeed. But then I put some more thinking into it. The PieImg class's solely purpose is to define js API for the custom element <pie-img>. This class should never be used in a different context than creating an html element. It should also not be imported twice (once the element is declared, it's ready to use anywhere in html and that's it). So it seems more reasonable to have the element register itself at import time. It is not so much a side effect here, as rather the core intention of importing this file in the first place - to be used in html. This is also what our canonically implemented elements are doing - check e.g. https://github.com/GoogleChromeLabs/howto-components/blob/master/elements/howto-checkbox/howto-checkbox.js

Reply:
I'm somewhat divided on this issue. On the one hand, letting a module make direct changes to the global namespace seems like a violation of encapsulation, particularly since tooling and architecture may lead to the same module being instanced more than once, despite our best efforts. On the other hand, I do agree that it seems more like a module concern, rather than a concern for the importer.

I'd say lets make the change for now, and re-evaluate if it leads to problems later. If you're making this change, you'll also need to modify `pie-item.test.js`, though, since it shouldn't try to define the custom element itself, like it's currently doing.